cinema/serializers.py

```diff
@@ -68,13 +78,77 @@
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available"
         )
 
 
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = serializers.SlugRelatedField(
+        source="tickets",
+        read_only=True,
+        many=True,
+        slug_field="full_name",
+    )
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = ("id", "show_time", "movie", "cinema_hall", "taken_places")
+
+
+class TicketSerializer(serializers.ModelSerializer):
+    def validate(self, attrs):
+        data = super(TicketSerializer, self).validate(attrs)
+        Ticket.validate_seat(
+            attrs["seat"],
+            attrs["row"],
+            attrs["movie_session"],
+            serializers.ValidationError
+        )
+        return data
```

Comment:
The `validate` method in `TicketSerializer` calls `Ticket.validate_seat`. Ensure that the `attrs` dictionary contains all necessary keys (`seat`, `row`, `movie_session`) before calling `validate_seat`, and handle any potential KeyError exceptions.